Bugsnag/Breadcrumbs/BugsnagBreadcrumbs.m

```diff
@@ -59,14 +55,16 @@ - (void)addBreadcrumb:(NSString *)breadcrumbMessage {
 
 - (void)addBreadcrumbWithBlock:
     (void (^_Nonnull)(BugsnagBreadcrumb *_Nonnull))block {
-    if (self.capacity == 0) {
+    if (self.config.maxBreadcrumbs == 0) {
         return;
     }
     BugsnagBreadcrumb *crumb = [BugsnagBreadcrumb breadcrumbWithBlock:block];
 
     if (crumb != nil && [self shouldSendBreadcrumb:crumb]) {
-        [self resizeToFitCapacity:self.capacity - 1];
         dispatch_barrier_sync(self.readWriteQueue, ^{
+            while ([self.breadcrumbs count] >= self.config.maxBreadcrumbs) {
```

Comment:
This will crash if `maxBreadcrumbs == 0`

Reply:
This is excluded above (L58).

Reply:
In fact, it makes me nervous in case the count is zero, so I've removed the loop (it will never be more than the max) and added a check for count>0.